libs/platform/src/lib/table/table.component.ts

```diff
@@ -1075,6 +1075,23 @@ export class TableComponent<T = any> extends Table<T> implements AfterViewInit,
         }
     }
 
+    /**
+     * Sets the checked state of the row.
+     * @param rowIndex Index of the row.
+     * @param value Value of the checked state.
+     */
+    setRowCheckedState(rowIndex: number, value: boolean): void {
```

Comment:
I think we can also expend this functionality just like we did with `rowNavigatable`, where the data of the row itself has `checked` flag, property name of which is set via @Input property of the table. This way developers could use just a datasource to mark the checked/unchecked row instead of iterating over the rows

Reply:
@platon-rov is addressing that with #9304